stdlib/importlib/metadata/__init__.pyi

```diff
@@ -170,15 +176,15 @@ class DistributionFinder(MetaPathFinder):
 
 class MetadataPathFinder(DistributionFinder):
     @classmethod
-    def find_distributions(cls, context: DistributionFinder.Context = ...) -> Iterable[PathDistribution]: ...
+    def find_distributions(cls, context: DistributionFinder.Context = ...) -> Iterable[PathDistribution[_SimplePath]]: ...
     if sys.version_info >= (3, 10):
         # Yes, this is an instance method that has argumend named "cls"
         def invalidate_caches(cls) -> None: ...
 
-class PathDistribution(Distribution):
-    def __init__(self, path: Path) -> None: ...
-    def read_text(self, filename: StrPath) -> str: ...
-    def locate_file(self, path: StrPath) -> PathLike[str]: ...
+class PathDistribution(Distribution, Generic[_TSimplePath]):
+    def __init__(self, path: _TSimplePath) -> None: ...
+    def read_text(self, filename: str) -> str: ...
+    def locate_file(self, path: _TSimplePath) -> _TSimplePath: ...  # type: ignore[override]
```

Comment:
This seems incorrect. It implies that a PathDistribution constructed with a Path as the argument to `__init__` will only accept a Path as the argument to `locate_file`, but that's not right:

```
In [47]: p = importlib.metadata.PathDistribution(Path("x"))

In [48]: p.locate_file("y")
Out[48]: PosixPath('y')
```

Reply:
I'm referring to the fact that I passed a string, not the Path/PosixPath change.

Reply:
whoops, didn't see you replied and deleted the previous message. Yes, I realised afterwards 😅